src/creatwth.cpp

```diff
@@ -499,6 +499,33 @@ static BOOL UpdateFrom32To64(HANDLE hProcess, HMODULE hModule, WORD machine,
 }
 #endif // DETOURS_64BIT
 
+typedef BOOL(WINAPI *LPFN_ISWOW64PROCESS)(HANDLE, PBOOL);
+
+static BOOL IsWow64ProcessHelper(HANDLE hProcess,
+                                 PBOOL Wow64Process)
+{
+    if (Wow64Process == NULL) {
+        return FALSE;
+    }
+
+    // IsWow64Process is not available on all supported versions of Windows.
+    // Use GetModuleHandle to get a handle to the DLL that contains the function
+    // and GetProcAddress to get a pointer to the function if available.
+
+    HMODULE hKernel32 = GetModuleHandleW(L"KERNEL32.DLL");
+    if (hKernel32 == NULL) {
+      return FALSE;
```

Comment:
Should we add  DETOUR_TRACE(...); messages on the failure paths to aid folks who might be debugging?

Reply:
I think under normal circumstances, this call to GetModuleHandleW can never fail. However I know there are some who use Detours (with modifications) in a context where only ntdll.dll is loaded, so I was being cautious. But yes, I agree and will add DETOUR_TRACE.

Reply:
Agreed, I was more worried about the GetProcAddress() failure, if they are on a downlevel platform. I meant to call out that we should trace both.

Reply:
Ah, I see. Sure, I will add that as well.